internal/controller/entra/main.go

```diff
@@ -184,3 +186,47 @@ func (c *Controller) AddtoInvitedUserGroup(ctx context.Context, email string) er
 	return err
 
 }
+
+// FindUsernames searches entra for usernames given a query of email, user principal or display name
+func (c *Controller) FindUsernames(ctx context.Context, query string) ([]types.Username, error) {
+	queryRegex := regexp.MustCompile(`^\w[\w.\s0-9@]+\w$`)
+	queryIsValid := queryRegex.MatchString(query)
+	if !queryIsValid {
+		return nil, types.NewErrInvalidObject("invalid query")
+	}
+
+	filterQuery := fmt.Sprintf(
+		`startswith(displayName,'%s') or startswith(userPrincipalName,'%s') or startswith(givenName,'%s') or startswith(mail,'%s')`,
+		query, query, query, query,
+	)
+	resultLimit := int32(20) // limits to 20 results or 20 per page if more than 20 results
+
+	headers := abstractions.NewRequestHeaders()
+
+	configuration := &graphusers.UsersRequestBuilderGetRequestConfiguration{
+		Headers: headers,
+		QueryParameters: &graphusers.UsersRequestBuilderGetQueryParameters{
+			Filter: &filterQuery,
+			Top:    &resultLimit,
+		},
+	}
+
+	data, err := c.client.Users().Get(ctx, configuration)
+	if err != nil {
+		return nil, err
+	}
+
+	userMatches := data.GetValue()
+	if userMatches == nil {
+		return nil, types.NewNotFoundError("no users matching query found in tenant")
+	} else {
+	}
```

Comment:
```suggestion
	if userMatches == nil || len(userMatches) == 0 {
		return nil, types.NewNotFoundError("no users matching query found in tenant")
	}
```

Reply:
question here about how we want to handle empty responses – `404` or `200` with empty slice?